python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -0,0 +1,253 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from collections.abc import AsyncGenerator
+from typing import Any
+
+from azure.ai.inference import load_client as load_client_sync
+from azure.ai.inference.aio import ChatCompletionsClient as ChatCompletionsClientAsync
+from azure.ai.inference.models import (
+    AssistantMessage,
+    AsyncStreamingChatCompletions,
+    ChatChoice,
+    ChatCompletions,
+    ChatRequestMessage,
+    ModelInfo,
+    ModelType,
+    StreamingChatChoiceUpdate,
+    SystemMessage,
+    ToolMessage,
+    UserMessage,
+)
+from azure.core.credentials import AzureKeyCredential
+from pydantic import ValidationError
+
+from semantic_kernel.connectors.ai.azure_ai_inference import (
+    AzureAIInferenceChatPromptExecutionSettings,
+    AzureAIInferenceSettings,
+)
+from semantic_kernel.connectors.ai.chat_completion_client_base import ChatCompletionClientBase
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.contents.chat_message_content import ChatMessageContent
+from semantic_kernel.contents.streaming_chat_message_content import StreamingChatMessageContent
+from semantic_kernel.contents.utils.author_role import AuthorRole
+from semantic_kernel.contents.utils.finish_reason import FinishReason
+from semantic_kernel.exceptions.service_exceptions import ServiceInitializationError
+
+_MESSAGE_CONVERTER: dict[AuthorRole, Any] = {
+    AuthorRole.SYSTEM: SystemMessage,
+    AuthorRole.USER: UserMessage,
+    AuthorRole.ASSISTANT: AssistantMessage,
+    AuthorRole.TOOL: ToolMessage,
+}
+
+
+class AzureAIInferenceChatCompletion(ChatCompletionClientBase):
+    """Azure AI Inference Chat Completion Service."""
+
+    client: ChatCompletionsClientAsync
+
+    def __init__(
+        self,
+        api_key: str | None = None,
+        endpoint: str | None = None,
+        service_id: str | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = None,
+        client: ChatCompletionsClientAsync | None = None,
+        model_info: ModelInfo | None = None,
+    ) -> None:
+        """Initialize the Azure AI Inference Chat Completion service.
+
+        Args:
+            api_key: The API key for the Azure AI Inference service deployment.
+            endpoint: The endpoint of the Azure AI Inference service deployment.
+            service_id: Service ID for the chat completion service. (Optional)
+            env_file_path: The path to the environment file. (Optional)
+            env_file_encoding: The encoding of the environment file. (Optional)
+            client: The Azure AI Inference client to use. (Optional)
+            model_info: The model info of the provided client. (Optional, required if client is provided)
+        """
+        if client is not None:
```

Comment:
this will probably be easier to read if we inverse the logic, so `if not client: #load from env, create_client`, and then adapt the _create_client (and maybe rename) to optionally create and validate, against model_info from a client and the supplied ai_model_id and model_provider, but I assume that if you supply the endpoint and key and model_name that it can work already or do you also need to specify the provider?

Reply:
Provider is not needed. We only need the model name since it's part of the Pydantic model. Model name is not required to create a client.